lib/ddtrace/utils.rb

```diff
@@ -1,9 +1,21 @@
+require 'thread'
+
 module Datadog
-  # TODO[manu]: write docs
+  # Utils contains low-level utilities, typically to provide pseudo-random trace IDs.
   module Utils
+    # We use a custom random number generator because we want no interference
+    # with the default one. Using the default prng, we could break code that
+    # would rely on srand/rand sequences.
+    @rnd = Random.new
```

Comment:
@ufoot yes my point is generic and related to a testing environment where the sequence can be reproducible